tests/src/main/scala/BasicQueryTests.scala

```diff
@@ -259,33 +264,31 @@ final class BasicJoinQueryTests(implicit xa: Transactor[IO]) {
   val debugOn = false
   def debug[A]: Pipe[IO, A, A] = if (debugOn) _.evalTap(s => IO(println(s))) else identity
 
-  def expectAllToBe[A[_[_]], Queried: Eq: Show](
+  def expectAllToBe[A[_[_]], Queried: Show](
     q: Query[A, Queried]
   )(
     first: Queried,
     rest: Queried*
   )(
-    implicit xa: Transactor[IO],
-    file: sourcecode.File,
-    line: sourcecode.Line
-  ): IO[Assertions] = {
+    implicit xa: Transactor[IO]
+  ): IO[NonEmptyList[Assertion]] = {
     val expectedList = first :: rest.toList
     val showQuery = (if (debugOn) IO(println(show"Testing query: ${q.compileSql.sql}")) else IO.unit)
 
     showQuery *> q.compileSql.stream.transact(xa).through(debug).compile.toList.attempt.map {
       case Left(exception) =>
-        Assertions(
-          Assertion.Failed(
-            AssertionFailure(
-              show"""An exception occured, but $expectedList was expected.
+        Assertion
+          .Failed(
+            show"""An exception occured, but $expectedList was expected.
               |Relevant query: ${pprint.apply(q).render}${scala.Console.RED}
```

Comment:
Flawless will need to adopt a model of colors that makes sure to get back to the color if substrings change it themselves.

Reply:
This also concerns diffx.